src/group_theory/solvable.lean

```diff
@@ -331,3 +334,127 @@ lemma is_simple_group.comm_iff_is_solvable :
 end⟩
 
 end is_simple_group
+
+section perm_not_solvable
+
+lemma not_solvable_of_mem_derived_series {g : G} (h1 : g ≠ 1)
+  (h2 : ∀ n : ℕ, g ∈ derived_series G n) : ¬ is_solvable G :=
+mt (is_solvable_def _).mp (not_exists_of_forall_not
+  (λ n h, h1 (subgroup.mem_bot.mp ((congr_arg (has_mem.mem g) h).mp (h2 n)))))
+
+/-- A type with 5 terms -/
+inductive weekday : Type
```

Comment:
I think this should be namespaced.

Reply:
Could you clarify what you mean by this? Do you mean that I should enclose the whole weekday stuff in another namespace. If so, then what should be done with the current weekday namespace that starts after the definition?

Reply:
The type would change to `foo.weekday` and the namespace would change to `foo.weekday` to match. `foo` would probably be something about `s5` and `solvable.